src/highlight-component.js

```diff
@@ -0,0 +1,60 @@
+function HighlightComponent() {
+  this.state = false;
+  this.components = extractComponentsFromPage();
+
+  function extractComponentsFromPage() {
+    return $('[class*="app-c"], [class*="pub-c"], [class*="govuk"]')
+      .toArray()
+      .reduce(function(array, element) {
+        var blockRegex = /(app-c-|pub-c-|govuk-c-|govuk-)([^ _\n]*(?=[ \n]|$))/;
+        var match = $(element).attr('class').match(blockRegex);
+
+        if (match) {
+          array.push({
+            element: element,
+            prefix: match[1],
+            name: match[2]
+          });
+        }
+
+        return array
+      }, []);
+  }
+
+  this.components.forEach(setupComponent.bind(this));
+
+  function setupComponent(component) {
+    var $element = $(component.element);
+    $element.attr('data-component-name', component.name);
```

Comment:
Could the name and prefix show? eg `pub-c-button` and `app-c-figure`, as the prefixes give an indication of where something lives without having to follow a link.

Reply:
👍